cbrain_task/fsl_melodic/bourreau/fsl_melodic.rb

```diff
@@ -50,82 +51,448 @@ def job_walltime_estimate #:nodoc:
+    cmds = []
+    cmds << "# Sets option ${parameter_name} in the design file\n"
+    cmds << sed_design_file(design_file_path,"\'set.*fmri(#{parameter_name})\'","\'# Line commented by CBRAIN set fmri(#{parameter_name})\'")
+    cmds << add_line_to_file(design_file_path,"'set fmri(#{parameter_name})' #{value}")
+    cmds << "\n"
+    return cmds.join
+  end
+  
+  ####
+  #### Conversion from MINC to Nifti.
+  ####
+  
+  # Returns the conversion command from MINC to NIFTI.
+  def mnc_to_nifti_command minc_file_name    
+    raise "Error: this doesn't look like a MINC file" unless is_minc_file_name? minc_file_name
+    nii_file_name = nifti_file_name minc_file_name
+    # removes the minc file after conversion otherwise feat crashes...
+    cmds = []
+    cmds << "# File converstion to Nifti\n"
+    cmds << "echo Converting file #{minc_file_name} to Nifti\n"
+    cmds << "mnc2nii -nii #{minc_file_name} `pwd`/#{File.basename nii_file_name}\n"
+    cmds << "if [ $? != 0 ]\n"
+    cmds << "then\n"
+    cmds << "  echo ERROR: cannot convert file #{minc_file_name} to nii\n"
+    cmds << "  exit 1\n"
+    cmds << "fi\n"
+    cmds << "rm -f #{minc_file_name}\n"
+    cmds << "\n"
+    return cmds.join
+  end
+  
+  # Gets a NIFTI file name from a MINC file name. 
+  def nifti_file_name minc_file_name
+    raise "Error: this doesn't look like a MINC file" unless is_minc_file_name? minc_file_name
+    name ="#{File.dirname(minc_file_name)}/#{File.basename(minc_file_name,".mnc")}" 
+    return "#{name}.nii"
+  end
+  
+  # Tests if file name looks like a MINC file. 
+  def is_minc_file_name? file_name
+    return file_name.end_with? ".mnc"
+  end
+  
+  # Returns an array containing the cache full name of the file converted to MINC
+  # format, and the conversion command.
+  def converted_file_name_and_command file_id
+    return nil if not file_id.present?
+    file_name = "#{self.full_cluster_workdir}/#{Userfile.find(file_id).name}"    
+    return [file_name,""] unless is_minc_file_name? file_name
+    # file_name is a MINC file name
+    return [ nifti_file_name(file_name) ,  mnc_to_nifti_command(file_name) ] 
+  end
 
-    self.addlog_to_userfiles_these_created_these( [ functional_file,structural_file ], [ outputfile ] )
-    self.addlog("Saved result file #{params[:output_dir_name]}")
-    params[:outfile_id] = outputfile.id
-    outputfile.move_to_child_of(functional_file)
+  ####
+  #### Other utils.
+  ####
+  
+  def modify_file_path_for_vm path
+    return nil if not path.present?
+    task_dir = RemoteResource.current_resource.cms_shared_dir
+    return path.sub(task_dir,File.join("$HOME",File.basename(task_dir)))
+  end
 
-    return true
+  # Returns a file_name if there is no file named file_name. Returns a
+  # unique file name based on file_name otherwise.
+  def unique_file_name file_name
+    count = 0
+    name  = file_name
+    while not Userfile.where(:name => name).empty?
```

Comment:
This seems to be used to make sure we have a unique file name to register a file later on (putting aside the obvious race condition). The userfiles this method queries for aren't used (thus maybe use exists? instead?).
For this method to work, all userfiles need to be checked, whether they belong to the current user or not.